Mm 46549 gather intent cloud monthly subscription

Afsoon opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
Allow the customers to submit their feedback for new payment options while they are buying their monthly subscription

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-server/issues/XXXXX

Otherwise, link the JIRA ticket.
-->

https://mattermost.atlassian.net/browse/MM-46549

#### Related Pull Requests
<!--
List all PRs related to resolving a ticket. For instance, if you submitted a PR to `mattermost/mattermost-server`, please include it here.
-->
- [Has server changes](https://github.com/mattermost/mattermost-server/pull/21410)

#### Screenshots
![image](https://user-images.githubusercontent.com/713534/195882104-59eff745-9ecf-49d6-ad7b-504939497382.png)

Stripe metadata
![image](https://user-images.githubusercontent.com/713534/195887329-a12c2653-018f-4023-bbaf-30892fff06ea.png)


#### Release Note
```release-note
Allow customers to submit their feedback for wire transfers to buy monthly subscription.
```

mattermod commented:
E2E tests not automatically triggered, because PR has no approval yet. Please ask a developer to review and then try again to attach the QA label.

Afsoon commented on components/widgets/modals/full_screen_modal.tsx:
```diff
@@ -47,7 +48,8 @@ class FullScreenModal extends React.PureComponent<Props> {
     private handleKeypress = (e: KeyboardEvent) => {
-        if (e.key === 'Escape' && this.props.show) {
+        const currentActiveElement = document.activeElement;
+        if (e.key === 'Escape' && this.props.show && e.target && this.modal.current && this.modal.current.contains(currentActiveElement)) {
```
This change is to close this modal **only** when the modal is focused.

Afsoon commented on components/widgets/modals/full_screen_modal.tsx:
```diff
@@ -21,20 +21,21 @@ type Props = {
     public componentDidMount() {
         document.addEventListener('keydown', this.handleKeypress);
-        document.addEventListener('focus', this.enforceFocus, true);
+        document.addEventListener('focus', this.enforceFocus, this.props.overrideTargetEvent);
```
From [MDN Docs](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener): if you pass a boolean and it's true, any focus event will be override by this listener. On our situation I had to disable it because it was impossible to focus on our textarea.

Afsoon commented:
95% of this PR are the same changes that from Portal.

neallred commented on components/payment_form/gather_intent/gather_intent.scss:
```diff
@@ -0,0 +1,171 @@
+
+        h3 {
+            margin: 0;
+            color: rgba(63, 67, 80, 1);
```
I think instances of 63, 67, 80 can be replaced with ` --center-channel-text-rgb`.

neallred commented on components/payment_form/gather_intent/gather_intent.scss:
```diff
@@ -0,0 +1,171 @@
+
+        &__textarea {
+            width: 100%;
+            border: solid 1px rgba(63, 67, 80, 0.16);
```
I think we'll also want to explicitly set the background for the textarea to `var(--center-channel-bg)` so that in dark mode it looks like the inputs on the form that opens this modal:
![dark-theme-pre](https://user-images.githubusercontent.com/13738432/195893218-8bebb919-2e7a-483c-8db5-9d5ad273c20e.png)

Success modal looks good to me in dark mode.

Afsoon replied:
Another test case to add to my list, thanks!.

AGMETEOR replied:
The font color on the dark mode theme still seems off. Can use `rgba(var(--center-channel-color-rgb), 0.72)`
<img width="1792" alt="Screenshot 2022-10-19 at 19 17 11" src="https://user-images.githubusercontent.com/28563179/196747842-eb68c09f-b373-4f73-91fc-5e127b132fb3.png">

neallred commented on utils/test_helper.ts:
```diff
@@ -438,4 +438,42 @@ export class TestHelper {
+            email: '',
+            contact_first_name: '',
+            contact_last_name: '',
+            create_at: new Date(Date.now()).getUTCDate(),
```
I'd recommend having create_at be `0` so that its in line with other mock functions in this helper, and so that it's easier to use if you want to use it to help generate an expected value that you will be using for checking deep equality.

neallred reviewed (approved):
Code from CWS looks good. webapp specific code looks good overall, had a few nonblocking nits. The only thing I think *needs* to change is the nit about having the textarea background be themed. Nice work, @Afsoon !

Afsoon commented on components/pricing_modal/content.tsx:
```diff
@@ -245,7 +245,7 @@ function Content(props: ContentProps) {
                             action: () => openPurchaseModal('click_pricing_modal_professional_card_upgrade_button'),
                             text: formatMessage({id: 'pricing_modal.btn.upgrade', defaultMessage: 'Upgrade'}),
-                            disabled: !isAdmin || isProfessional || (isEnterprise && !isEnterpriseTrial),
+                            disabled: false,
```
Yes, I have to revert it. nice catch!

Afsoon commented:
/e2e-test

mattermod commented:
Successfully triggered e2e testing!
https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/251034

yasserfaraazkhan commented:
@Afsoon add a e2e test with lint fixes here https://github.com/mattermost/mattermost-webapp/pull/11371

AGMETEOR reviewed (commented):
Just a small fix to make with the modal. See here https://github.com/mattermost/mattermost-webapp/pull/11360#discussion_r999684301

Afsoon commented:
Added the suggestion for CSS also updated the CSS and changes made in the other PR 

<img width="627" alt="image" src="https://user-images.githubusercontent.com/713534/197507005-4fd8f609-5631-4181-ad9b-2f5868831437.png">

Afsoon commented:
gentle ping @AGMETEOR

marianunez commented:
@yasserfaraazkhan is this QA Review approved? I noticed you changed the tag but there is no approval or comments so double checking here

yasserfaraazkhan commented:
> @yasserfaraazkhan is this QA Review approved? I noticed you changed the tag but there is no approval or comments so double checking here

This change is verified @Afsoon @marianunez. Added a e2e and also verified the manual scenarios mentioned in ticket

AGMETEOR reviewed (approved):
LGTM
